helix-core/src/line_ending.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{Rope, RopeGraphemes, RopeSlice};
+
+/// Represents one of the valid Unicode line endings.
+#[derive(PartialEq, Copy, Clone, Debug)]
+pub enum LineEnding {
+    Crlf, // CarriageReturn followed by LineFeed
+    LF,   // U+000A -- LineFeed
+    CR,   // U+000D -- CarriageReturn
+    Nel,  // U+0085 -- NextLine
+    LS,   // U+2028 -- Line Separator
+    VT,   // U+000B -- VerticalTab
+    FF,   // U+000C -- FormFeed
+    PS,   // U+2029 -- ParagraphSeparator
+}
+
+pub fn rope_slice_to_line_ending(g: &RopeSlice) -> Option<LineEnding> {
+    if let Some(text) = g.as_str() {
+        str_to_line_ending(text)
+    } else if g == "\u{000D}\u{000A}" {
+        Some(LineEnding::Crlf)
+    } else {
+        // Not a line ending
+        None
+    }
+}
+
+pub fn str_to_line_ending(g: &str) -> Option<LineEnding> {
```

Comment:
I think you can just include all the line endings here as well.  We will probably want to use this function in other parts of Helix as well, where we'll want to detect all possible line endings.  (In fact, I'm working on another PR where I'll want exactly that.)

Let's reject VT, FF, and PS in the auto-detect code itself, so that this can remain a more general utility function.